### PR TITLE
Headless/tahsin/compare/issue fix

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **Jira** task here for reference.
- [ ] Bug fix
- [ ] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [ ] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/headless/src/modules/order/services/customer.service.ts [hunk 1]
```diff
@@ -33,45 +38,46 @@ export class OrderCustomerService {
     );
   }
 
-  async reOrder(userId: string, body: ReOrderQuery): Promise<IServiceResponse<CartResponse>>{
+  async reOrder(userId: string, body: ReOrderQuery): Promise<ReOrderResponse>{
     let { ignoreInvalidItems, overWriteCart, orderId } = body; 
 
     const prevOrder = await this.orderRepository.findOrder({orderId, userId});
-    if(!prevOrder) return errorResponse( ErrorMessageReOrder.INVALID_ID,null,HttpStatus.BAD_REQUEST );
+    if(!prevOrder) return {error: ErrorMessageReOrder.INVALID_ID,errors: null, code: HttpStatus.BAD_REQUEST};
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/order/services/customer.service.ts:45]: what was the problem here? i mean why this change is needed?

File: packages/models/src/order/reOrder.interface.ts [hunk 2]
```diff
@@ -6,4 +9,23 @@ export const enum ErrorMessageReOrder{
...
     ALL_ITEMS_INVALID = 'THESE ITEMS ARE NOT AVAILABLE RIGHT NOW',
     CANNOT_ADD_ITEMS = 'CANNOT ADD ITEMS TO THE CART'
-}
\ No newline at end of file
+}
+export interface ReOrderData {
+    id?: string;
+    userId?: string;
+    products?: IOrderProduct[] | null;
+    reDirectHome?: boolean;
```
Comment (Anjarul-Islam) [inline packages/models/src/order/reOrder.interface.ts:17]: why this flag (reDirectHome) is needed?

Comment (ghost) [reply packages/models/src/order/reOrder.interface.ts:17]: this flag is needed for frontend as they requested

File: packages/headless/src/modules/order/services/customer.service.ts (hunk 1 above)
Comment (ghost) [reply packages/headless/src/modules/order/services/customer.service.ts:45]: as the user is not requesting for anything invalid initiaaly, so it shouldn't be a bad request or any kind of error response

Comment (ghost) [reply packages/headless/src/modules/order/services/customer.service.ts:45]: also the frontend team was unable to retrieve the error as it was getting caught in catch block

**Comment by ghost** [comment]
[https://pm23.atlassian.net/browse/BC-567](url)
